packages/discord.js/src/managers/GuildMemberRoleManager.js

```diff
@@ -174,6 +154,53 @@ class GuildMemberRoleManager extends DataManager {
     }
   }
 
+  /**
+   * @typedef {Object} ModifyGuildMemberRolesOptions
+   * @property {Readonly<RoleResolvable[]> | ReadonlyCollection<Snowflake, Role>} [rolesToAdd] The roles to add
+   * @property {Readonly<RoleResolvable[]> | ReadonlyCollection<Snowflake, Role>} [rolesToRemove] The roles to remove
+   * @property {Readonly<RoleResolvable[]> | ReadonlyCollection<Snowflake, Role>} [reason] Reason for modifying
+   * the roles
+   */
+
+  /**
+   * Modifies the roles of the member.
+   * @param {ModifyGuildMemberRolesOptions} [options] Options for modifying the roles
+   * @returns {GuildMember}
+   */
+  modify(options) {
+    const resolvedRolesToAdd = this.resolveRoles(options.rolesToAdd);
+    const resolvedRolesToRemove = this.resolveRoles(options.rolesToRemove);
+
+    const currentRoles = new Set(this.member.roles.cache.keys());
+    for (const role of resolvedRolesToAdd) {
+      currentRoles.add(role.id);
+    }
+    for (const role of resolvedRolesToRemove) {
+      currentRoles.delete(role.id);
+    }
+
+    return this.member.roles.set([...currentRoles], options.reason);
+  }
+
+  /**
+   * Resolves roles from the input.
+   * @param {Readonly<RoleResolvable[]> | ReadonlyCollection<Snowflake, Role>} rolesToResolve The roles to resolve
+   * @returns {Role[]} The resolved roles
```

Comment:
this is not what is returned by this function.

an array of role ids is returned.

as this is only used to resolve role ids and is private we should keep it as returning role ids as makes more sense. This will require a small change to the mosify function as well.